jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/IRabitTracker.java

```diff
@@ -0,0 +1,20 @@
+package ml.dmlc.xgboost4j.java;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface for Rabit tracker implementations with three public methods:
+ *
+ *  - start(): Start the rabit tracker awaiting for worker connections.
+ *  - getWorkerEnvs(): Return the environment variables needed to initialize Rabit clients.
+ *  - waitFor(): Wait for the task execution by the worker nodes.
+ */
+public interface IRabitTracker {
+  Map<String, String> getWorkerEnvs();
+  boolean start();
+  // for Scala-implementation compatibility
+  boolean start(long timeout, TimeUnit unit);
+  int waitFor();
+  int waitFor(long timeout, TimeUnit unit);
```

Comment:
I removed the first one from the `start()` method, which is now in the constructor with a default value. 

The third one (connection timeout) is actually highly dependent on the training data size. 
Because the tracker is started before calling `buildDistributedBoosters`, and the `mapPartitions` in that function is dependent on data loading, which is lazily executed. 
Larger datasets take over 10 minutes to load, and therefore the Rabit workers in `mapPartitions` will start to connect 10 minutes after the tracker has been started. 

As for the second one, I think syntactically it looks more intuitive to have `waitFor()` with a duration argument, instead of specifying it in the constructor. 

To simplify the interface under these considerations, I only kept `start(workerConnectionTimeoutMillis)` and `waitFor(taskExecutionTimeoutMillis)` in `IRabitTracker` interface, and if the timeout is zero, it waits indefinitely.